docs/content/getting-started/project-file-reference.mdx

```diff
@@ -0,0 +1,438 @@
+        <a href="/concepts/code-locations">Code locations documentation</a> to
+        learn more.
+        <br />
+        <br /> <strong>Note:</strong> <code>pyproject.toml</code> was introduced
+        in{" "}
+        <a href="https://peps.python.org/pep-0518/https://peps.python.org/pep-0518/">
+          PEP-518
+        </a>{" "}
+        and meant to replace <code>setup.py</code>, but we may still include a{" "}
+        <code>setup.py</code> for compatibility with tools that do not use this
+        spec.
+      </td>
+    </tr>
+    <tr>
+      <td>setup.py</td>
+      <td>
+        A build script with Python package dependencies for your new project as
+        a package.
+        <br /><br />
+        <strong>Note</strong>: If using Dagster Cloud, <code>dagster-cloud</code> as a dependency.
+      </td>
+    </tr>
+    <tr>
+      <td>setup.cfg</td>
+      <td>
+        An ini file that contains option defaults for <code>setup.py</code>{" "}
+        commands.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+---
+
+## Configuration files
+
+<Note>
+  <strong>Not sure where to put these files?</strong> Unless otherwise noted,
+  all configuration files should be placed in the root of your Dagster project
+  directory.
+</Note>
+
+Depending on your use case or if you're using Dagster Cloud, you may also need to add additional configuration files to your project:
+
+<table
+  className="table"
+  style={{
+    width: "100%",
+  }}
+>
+  <thead>
+    <tr>
+      <th
+        style={{
+          width: "20%",
+        }}
+      >
+        File/Directory
+      </th>
+      <th>Description</th>
+      <th
+        style={{
+          width: "10%",
+        }}
+      >
+        OSS
+      </th>
+      <th
+        style={{
+          width: "10%",
+        }}
+      >
+        Cloud
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>dagster.yaml</td>
+      <td>
+        Configures your Dagster instance, including defining storage locations,
+        run launchers, sensors, and schedules. Refer to the{" "}
+        <a href="somelinkhere">dagster.yaml reference</a> for more info,
+        including a list of use cases and available options.
+        <br />
+        <br />
+        For Dagster Cloud <a href="/dagster-cloud/deployment/hybrid">
+          Hybrid deployments
+        </a>, this file can be used to <a href="/dagster-cloud/deployment/agents/customizing-configuration">
+          customize the Hybrid agent
+        </a>.
+      </td>
+      <td>Optional</td>
+      <td>Optional</td>
+    </tr>
+    <tr>
+      <td>dagster_cloud.yaml</td>
+      <td>
+        Defines multiple code locations for Dagster Cloud. For Hybrid
+        deployments, this file can be used{" "}
+        <a href="/dagster-cloud/managing-deployments/setting-environment-variables-dagster-cloud-agents">
+          to manage environment variables/secrets
+        </a>
+        .
+      </td>
+      <td>n/a</td>
+      <td>Optional</td>
```

Comment:
This file should be marked as Required. There are cases where the file is not strictly required (if you configure a code location through the UI or if you run `dagster-cloud` CLI commands with no file), but in practice I think every Cloud deployment uses this file and listing it as optional is confusing. If Required is too strong, than I'd favor Recommended over Optional.